keras_cv/layers/preprocessing/invert.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tensorflow as tf
+
+
+class Invert(tf.keras.layers.Layer):
+    """Inverts the image pixels.
```

Comment:
This is a somewhat ambiguous description, please provide the equation below. In particular, "invert" taking literally means "1 / x", but this is `255 - x`

Reply:
For what its worth, in the RandAugment linked below they distinguish it by applying a threshold to `Solarize`.  

https://github.com/tensorflow/similarity/blob/906d141b75acf00d0e1d37dd0d88c432bbb30f57/tensorflow_similarity/augmenters/img_augments.py#L449

That being said... the original `RandAugment` specification (https://arxiv.org/pdf/1909.13719.pdf) doesn't even include `Invert`.  Only `solarize`. 

AutoAugment does include a mention of `invert`, and defines the two operations as seen below:

<img width="506" alt="Screen Shot 2022-02-07 at 9 43 43 PM" src="https://user-images.githubusercontent.com/12191303/152925696-3bfc60f6-f043-4806-bac3-cea1084de7cd.png">

Reply:
Apart from what @LukeWood mentioned, `Invert` name is also used in most other implementations. It might not be the best, but changing it now could introduce confusion.

[Link 1](https://github.com/tensorflow/models/blob/dd02069717128186b88afa8d857ce57d17957f03/official/legacy/image_classification/augment.py#L520)
[Link 2](https://github.com/tensorflow/similarity/blob/906d141b75acf00d0e1d37dd0d88c432bbb30f57/tensorflow_similarity/augmenters/img_augments.py#L679)
[Link 3](https://imgaug.readthedocs.io/en/latest/source/overview/arithmetic.html#invert)

Reply:
In the official reference impl of the paper It Is just:
https://github.com/tensorflow/tpu/blob/master/models/official/efficientnet/autoaugment.py#L398-L401

In other Google papers is PIL/PILLOW `invert`

https://github.com/google-research/google-research/blob/master/ieg/dataset_utils/randaugment.py#L239:L241

Reply:
Precedent is one thing to look at, but correctness matters too. Deep learning researchers are memoryless and tend to be ignorant about established nomenclature, which leads to a lot of confusion for everyone down the road. A few examples from our recent past:

- What people call "normalization" in contexts such as "batch normalization" or "layer normalization" (which is an even more meaningless name...) is actually standardization. "Normalization" refers to processing data so that it has a certain norm (usually unit L2 norm). That's where the name "normalization" comes from. Seeking a fixed standard deviation is "standardization" (the name "standardization" comes "standard deviation"). There are no norms involved in batch normalization / etc, which should be called batch standardization.

- A majority of researchers are calling "dilated convolution" "atrous convolution". There are several layers of confusion here. "Atrous" is an English word which means "black". That's entirely unrelated to dilated convolution. The word seems to come from the French term "à trous" which literally means "with holes" and is used in French math textbooks to mean "sparse". But "sparse convolution" is convolution with sparse tensors, not dilated convolution (and if you really wanted to talk about sparse convolution, you'd call it sparse convolution, not "convolution à trous", because you're writing in English, not French). The fact that the Keras API went with the correct term "dilated convolution" played a meaningful role in fixing the nomenclature confusion here.

- Once upon a time most people were calling "transposed convolution" "deconvolution". Deconvolution is actually a different operation. Unfortunately the Keras API originally made the mistake of going with the mainsteam, and we had to change our layer names back to "Conv2DTranspose" later down the road.

- Etc.

In the current situation it does appear that the proper and specific term is solarization, not inversion. "Inversion" is super vague anyway. The first thing comes to mind when reading "inversion" is "taking the inverse" (1 / x). The second that comes to mind is "reversing", as in reversing the order of an input sequence. Etc.

Reply:
Well, one thing is clear to me:

Even if we wanted to match their terminology, we can still fit this need by just using a solarization layer with a threshold=None. Then it solarizes everything, executing the "Invert" operation.  So in this case we should go with a thresholded solarization layer.

Reply:
Agree with @LukeWood . Adding threshold here, and setting it to None sounds like a good compromise. I will be adding this.